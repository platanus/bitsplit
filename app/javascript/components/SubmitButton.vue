<template>
  <button @click="$emit('do-click')"
    :class= [buttonClass,colormod,hovermod,widemod]
    :disabled= fieldDisabled 
  >
    {{ fieldPlaceholder }}
  </button>
</template>
<script>
export default {
  name: 'SubmitButton',
  data () {
    return {
      colormod: 'bg-blue-500',
      hovermod: 'hover:bg-blue-700',
      widemod: 'w-normal',
      buttonClass: 'text-white font-bold py-2 px-4 rounded focus:outline-none focus:shadow-outline'
    }
  },
  props: {
    fieldDisabled: Boolean,
    fieldPlaceholder: String,
    color: String,
    wide: String
  },
  mounted: function() {
    if (this.wide == 'full') {
      this.widemod = 'w-full'
    }
    if (this.color == 'secondary') {
      this.colormod = 'bg-indigo-500'
      this.hovermod = 'hover:bg-indigo-700'
    }
  }
}
</script>
