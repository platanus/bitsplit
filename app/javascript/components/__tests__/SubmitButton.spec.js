import { shallowMount } from '@vue/test-utils';
import SubmitButton from '../SubmitButton';

describe('SubmitButton', () => {
  it('matches snapshot', () => {
    const wrapper = shallowMount(SubmitButton, {
      propsData: { fieldPlaceholder: 'Button' },
    });
    expect(wrapper).toMatchSnapshot();
  });
});
