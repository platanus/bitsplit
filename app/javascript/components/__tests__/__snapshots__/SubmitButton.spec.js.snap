// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SubmitButton matches snapshot 1`] = `<button class="text-white font-bold py-2 px-4 rounded focus:outline-none focus:shadow-outline bg-blue-500 hover:bg-blue-700 w-normal"></button>`;
